### DISCUSSION TITLE
Parsing a String to Float

### DISCUSSION THREAD

**Discussion by physics515** [rune-rs/rune#596] [Q&A] [answered]
Hello All,

Am I missing something? I've looked through the available docs and tried to make it up with every way that it makes sense to me but with no success.

The API returns the price as a JSON string. How do I get price to return as float?
``` rune
pub async fn price() {
    let response = http::get("https://midgard.thorswap.net/v2/pools").await?;
    let json = response.json().await?;
    let price = json[0]["assetPriceUSD"];

    price
}
```

Thank you for your help!

**Comment by udoprog** [comment, marked as answer]
Did you try?

```rust
let price = f64::parse(json[0]["assetPriceUSD"])?;
```

> Note that `float` has been renamed to `f64` in main, but not yet released.

**Reply by physics515** [reply]
I thought I did, but apparently it decided to work now.

Well, this worked:
``` rust
let price = float::parse(json[0]["assetPriceUSD"])?;
```

So float is becoming f64? i thought that it was the other way around...
